src/packageurl/contrib/url2purl.py

```diff
@@ -299,9 +299,8 @@ def build_cargo_purl(url):
 
 
 github_raw_content_pattern = (
-    r"^https?://raw.githubusercontent.com/"
-    r"(?P<namespace>.+)/(?P<name>.+)/(?P<version>.+)/"
-    r"(?P<subpath>.+)$"
+    r"https?://raw.githubusercontent.com/(?P<namespace>.+)/(?P<name>.+)/"
```

Comment:
@tdruez can you please have a look at the raw.githubusercontent.com URLs, I can't understand why this regex is not working

Reply:
In those cases, try to provide more context about what "is not working" so we identify the issue and get to a solution faster.

Here's some insight of my approach to figure out the issue starting with "https://raw.githubusercontent.com/LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as"

```
>>> url = "https://raw.githubusercontent.com/LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as"
>>> re.compile(r"https?://raw.githubusercontent.com/(?P<namespace>.+)/(?P<name>.+)/(?P<version>[^/]+)/(?P<subpath>.*)$", re.VERBOSE).match(url).groupdict()
{'namespace': 'LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx', 'name': 'containers', 'version': 'accordionClasses', 'subpath': 'AccordionHeader.as'}
```

We can see that the expected namespace is not captured properly as it is allowing the `/` in the match pattern. Looking at the `(?P<version>[^/]+)` segment, the `/` is excluded from the match for the version but missing from the `namespace` and the `name`.

A better pattern syntax could be: `r"https?://raw.githubusercontent.com/(?P<namespace>[^/]+)/(?P<name>[^/]+)/(?P<version>[^/]+)/(?P<subpath>.*)$"`

```
>>> url = "https://raw.githubusercontent.com/LeZuse/flex-sdk/master/frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as"
>>> re.compile(r"https?://raw.githubusercontent.com/(?P<namespace>[^/]+)/(?P<name>[^/]+)/(?P<version>[^/]+)/(?P<subpath>.*)$", re.VERBOSE).match(url).groupdict()
{'namespace': 'LeZuse', 'name': 'flex-sdk', 'version': 'master', 'subpath': 'frameworks/projects/mx/src/mx/containers/accordionClasses/AccordionHeader.as'}
```

Note that we would need a different approach if `/` exist in the name and namespace on that repo source.

Reply:
Got it, thanks :), will be making changes soon

Reply:
See also #53 for some ideas on how to simplify route declarations. Feedback welcome.